Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -310,34 +310,54 @@ BorderRadiiData PaintableBox::normalized_border_radii_data(ShrinkRadiiForBorders
     return border_radius_data;
 }
 
-void PaintableBox::before_children_paint(PaintContext& context, PaintPhase phase, ShouldClipOverflow should_clip_overflow) const
+Optional<Gfx::IntRect> PaintableBox::clip_rect() const
 {
-    if (!AK::first_is_one_of(phase, PaintPhase::Background, PaintPhase::Border, PaintPhase::Foreground))
-        return;
+    if (!m_clip_rect.has_value()) {
+        if (containing_block() && containing_block()->paint_box()) {
+            m_clip_rect = containing_block()->paint_box()->clip_rect();
+        }
```

Comment:
Little nit: Normally we don't bother with braces around the bodies of single-line ifs.